test-suite/tests/integration.rs

```diff
@@ -0,0 +1,73 @@
+use libtest_mimic::Arguments;
+use test_suite::{ERunner, RunnerTrait, TestConfig, TestManager};
+
+use std::{
+    fs,
+    net::{Ipv4Addr, SocketAddr, SocketAddrV4, TcpStream},
+    thread,
+    time::Duration,
+};
+
+// Read and parse the TOML file into a TestConfig
+fn read_test_cases() -> Result<TestConfig, anyhow::Error> {
+    let file_path = format!("{}/tests/tests.toml", env!("CARGO_MANIFEST_DIR"));
+    let file_content = fs::read_to_string(file_path).expect("File should be read without problems");
+    let test_config: TestConfig = toml::from_str(&file_content)?;
+    Ok(test_config)
+}
+
+fn wait_for_ports(ports: Vec<u16>) {
+    let ip = Ipv4Addr::new(127, 0, 0, 1);
+
+    let mut remaining_ports = ports;
+
+    while !remaining_ports.is_empty() {
+        remaining_ports.retain(|&port| {
+            let sockaddr = SocketAddr::new(SocketAddr::V4(SocketAddrV4::new(ip, port)).ip(), port);
+            match TcpStream::connect_timeout(&sockaddr, Duration::from_secs(3)) {
+                Ok(_) => {
+                    eprintln!("Connected to port {}", port);
+                    false
+                }
+                Err(_) => {
+                    eprintln!("Waiting for port {}...", port);
+                    thread::sleep(Duration::from_secs(3));
+                    true
+                }
+            }
+        });
+    }
+}
+
+fn main() -> Result<(), anyhow::Error> {
```

Comment:
nit #2: Since this entire code is meant as a test runner, we could make it a binary, move all of this to `main.rs`, and run it with `cargo run` instead of `cargo test`, which should produce similar results.

Reply:
But won't the convention of using it as `cargo test` conform to testing ?

Reply:
I added it to `main.rs`. Being in favour of `cargo run` with you